.jenkins

```diff
@@ -148,7 +148,7 @@ pipeline {
                                 -DCMAKE_BUILD_TYPE=Release \
                                 -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
                                 -DCMAKE_CXX_COMPILER=$WORKSPACE/bin/nvcc_wrapper \
-                                -DCMAKE_CXX_FLAGS=-Werror --Werror=all-warnings -Xcudafe --diag_suppress=3159 \
+                                -DCMAKE_CXX_FLAGS="-Werror --Werror=all-warnings -Xcudafe --diag_suppress=3159 -Xcudafe --diag_suppress=940" \
```

Comment:
I've added the `diag_suppress=940` because otherwise we get to fix:
> error #940-D: missing return statement at end of non-void function
in `Kokkos::DualView::view` method.

But it is not "trivial", because this method is `__host__ __device__`, and the usual `__builtin_unreachable` trick is `__host__`.

My first attempt was then to add `Kokkos::abort("This never happens.");` but I'm a bit reluctant adding this.

I find ignoring the `940` error code a good compromise.

Reply:
See discussion in https://github.com/kokkos/kokkos/pull/7455#discussion_r1804591172.